account_asset_disposal/models/account_asset_asset.py

```diff
@@ -111,6 +50,7 @@ def action_disposal_undo(self):
                 asset.state = 'close'
             else:
                 asset.state = 'open'
+                asset.method_number = asset.category_id.method_number
```

Comment:
Yes, but you can't change the currrent number. It can be customize for each asset. The value set is already correct.